### DISCUSSION TITLE
Rearrange Order of New Row Sidebar Panel

### DISCUSSION THREAD

**Discussion by mrnoisytiger** [Budibase/budibase#11660] [Help] [answered]
How would I go about rearranging the order of the new row sidebar panel that is built into the "Table Block" component? 

I set up my data set with the columns in a pretty lackluster order and I'm trying to go about rearranging them. I was able to rearrange within the actual data set itself and it shows up correctly on the "Data" tab. 

I'm using the Table Block component as my landing page and I'm able to create a row using the "Add Row" built-in button. However, the button is still reflecting the old column order, and not the new updated one. 

Any thoughts?

**Comment by andz-bb** [comment, marked as answer]
hey @mrnoisytiger

Currently the only way to rearrange the form fields in the table block side panels is to [eject the table block and form blocks](https://docs.budibase.com/docs/blocks#ejecting-blocks) so that they are split into individual form components. You can then rearrange them in the component tree as you can with other components.

As part of our August release (2.10) coming later this week, the table block has now been updated to allow you to choose and rearrange your form fields from within the block's settings. This will mean there's no longer any need to eject the blocks for this functionality. More info will be available on this very soon! 😃
